File: app/src/javaDev/java/com/example/ama/viewer/di/modules/OkHttpModule.java
package com.example.ama.viewer.di.modules;

import android.util.Log;

import com.example.ama.viewer.data.interceptor.RequestHeadersInterceptor;
import com.facebook.stetho.okhttp3.StethoInterceptor;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;

@Module
public class OkHttpModule {

    @Provides
    @Singleton
    OkHttpClient provideOkHttpClient(RequestHeadersInterceptor requestHeadersInterceptor,
                                     StethoInterceptor stethoInterceptor,
                                     HttpLoggingInterceptor loggingInterceptor) {
        return new OkHttpClient.Builder()
                .addInterceptor(requestHeadersInterceptor)
                .addInterceptor(loggingInterceptor)
                .addNetworkInterceptor(stethoInterceptor)
                .build();
    }

    @Provides
    @Singleton
    HttpLoggingInterceptor provideLoggingInterceptor() {
        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor(
                message -> Log.d("Logging interceptor", message));
        interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
        return interceptor;
    }
}

File: app/src/kotlinProd/java/com/example/ama/viewer/di/modules/OkHttpModule.kt
package com.example.ama.viewer.di.modules

import com.example.ama.viewer.data.interceptor.RequestHeadersInterceptor
import com.facebook.stetho.okhttp3.StethoInterceptor
import dagger.Module
import dagger.Provides
import okhttp3.OkHttpClient
import javax.inject.Singleton

@Module
class OkHttpModule {

    @Provides
    @Singleton
    fun provideOkHttpClient(requestHeadersInterceptor: RequestHeadersInterceptor,
                            stethoInterceptor: StethoInterceptor): OkHttpClient =
            OkHttpClient.Builder()
                    .addInterceptor(requestHeadersInterceptor)
                    .addNetworkInterceptor(stethoInterceptor)
                    .build()
}